dist/logs/logzioLogsFunction/core.js

```diff
@@ -1,60 +1,186 @@
-const logger = require('logzio-nodejs');
-const DataParser = require('./data-parser');
 
-function getCallBackFunction(context) {
-  return function callback(err, bulk) {
-    if (err) {
-      context.err(`logzio-logger error: ${err}`, err);
-      context.bindings.outputBlob = bulk;
+const { ContainerClient } = require('@azure/storage-blob');
+const logger = require('logzio-nodejs');
+const fs = require('fs');
+const util = require('util');
+const appendFileAsync = util.promisify(fs.appendFile);
+const containerName = 'logziologsbackupcontainer';
+const updateTimestamp = log => {
+    if (log.time) {
+        return {
+            ...log,
+            '@timestamp': log.time,
+        };
     }
-    context.done();
-  };
+    return log;
+};
+const updateFolderSize = () => {
+    const stats = fs.statSync(fileName);
+    const fileSizeInKb = stats.size / 1000;
+    folderSize += fileSizeInKb;
+}
+const arrayIsEmpty = array => {
+    return Array.isArray(array) && array.length == 0;
+};
+const getDate = () => {
+    const dateObj = new Date();
+    const month = dateObj.getUTCMonth() + 1; //months from 1-12
+    const day = dateObj.getUTCDate();
+    const year = dateObj.getUTCFullYear();
+    return year + '-' + month + '-' + day;
+};
+const uniqString = () => {
+    return Math.random()
+        .toString(36)
+        .substring(7);
+};
+const createFolderName = () => {
+    const newFolderName = getDate() + '-' + uniqString();
+    fs.mkdir(newFolderName, { recursive: true }, err => {
+        if (err) throw err;
+    });
+    return newFolderName;
+};
+let folderName = createFolderName();
+let fileName = '';
+const workingDir = process.cwd();
+const createFileName = (context) => {
+    fileName = 'logs-'.concat(uniqString()).concat('.txt');
+    return fileName;
+};
+const maxBatchSize = 100;
+const MB = 1000;
+let folderSize = 0;
+let logIndex = 1;
+const isObject = obj => {
+    return obj !== undefined && obj !== null && obj.constructor == Object;
+};
+const deleteEmptyFields = obj => {
+    for (const key in obj) {
+        if (isObject(obj[key])) {
+            deleteEmptyFields(obj[key]);
+        }
+        if (key === '') {
+            delete obj[key];
+        }
+    }
+};
+const getCallBackFunction = context => {
+    return function callback(err, bulk) {
+        if (err) {
+            context.err(`logzio-logger error: ${err}`, err);
```

Comment:
you're writing err twice, once in the message and once in the extra params, is it necessary?
are you doing something with the bulk?

Reply:
both were for old version - erasing bulk and changing context structure